Defining `HAVE_STDINT_H` globally can cause compile error

### Description

We are defining the `HAVE_STDINT_H` macro globally, when using the bundled CELT module:
https://github.com/mumble-voip/mumble/blob/7419dbf28899f64f8a6e5b59d33815776a845c06/src/mumble/CMakeLists.txt#L678-L681

However, other libraries can end up defining the same macro in their config headers (e.g. generated by autotools) in which case a redefinition warning is created. Since we are treating warnings as errors, that would fail to compile.

One example of where this can happen is with the `expat` library (used by `poco`) as was reported on our forums.

### Steps to reproduce

Not quite clear, but presumably one has to create a static build (on Windows) using a more recent vcpkg version that what we are currently using on CI.

### Mumble version

201dc2ea5b3ef2a69876f2dd81afa7d35e982f0e

### Mumble component

Client

### OS

Windows

### Reproducible?

Haven't tried

### Additional information

Original forums post: https://forums.mumble.info/topic/5721-windows-build-error/?do=findComment&comment=12226

Expat issue: https://github.com/libexpat/libexpat/issues/521

### Relevant log output

```shell
-DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_RESTRICTED_CAST_FROM_ASCII -DQT_SQL_LIB -DQT_SVG_LIB -DQT_THEME_SUPPORT_LIB -DQT_USE_FAST_CONCATENATION -DQT_USE_FAST_OPERATOR_PLUS -DQT_WIDGETS_LIB -DQT_WINDOWSUIAUTOMATION_SUPPORT_LIB -DQT_XML_LIB -DRESTRICT="" -DUNICODE -DUSE_GKEY -DUSE_JACKAUDIO -DUSE_MANUAL_PLUGIN -DUSE_OPUS -DUSE_OVERLAY -DUSE_PORTAUDIO -DUSE_QSSLDIFFIEHELLMANPARAMETERS -DUSE_RNNOISE -DUSE_WASAPI -DUSE_XBOXINPUT -DUSE_ZEROCONF -DWIN32_LEAN_AND_MEAN -DXML_STATIC -D_UNICODE -D_USE_MATH_DEFINES -D_WIN32_WINNT=0x0601 -Dmumble_EXPORTS -ID:\mumble\build\src\mumble\mumble_autogen\include -ID:\mumble\src\mumble -ID:\mumble\src\mumble\widgets -ID:\mumble\src -ID:\mumble\3rdparty\smallft -ID:\mumble\plugins -ID:\mumble\3rdparty\opus\include -ID:\mumble\3rdparty\celt-0.7.0-src\libcelt -ID:\mumble\3rdparty\qqbonjour -ID:\mumble\src\mumble\SYSTEM -ID:\mumble\3rdparty\jack -ID:\mumble\3rdparty\portaudio -ID:\mumble\build\src -ID:\mumble\3rdparty\xinputcheck-build\..\xinputcheck-src -ID:\mumble\3rdparty\speex-build\SYSTEM -ID:\mumble\3rdparty\speex-build\..\speex-src\include -ID:\mumble\3rdparty\speex-build\..\speexdsp\include -ID:\mumble\3rdparty\rnnoise-build\SYSTEM -ID:\mumble\3rdparty\rnnoise-build\..\rnnoise-src\include -ID:\vcpkg\installed\x64-windows-static-md\include -ID:\vcpkg\installed\x64-windows-static-md\include\qt5 -ID:\vcpkg\installed\x64-windows-static-md\include\qt5\QtCore -ID:\vcpkg\installed\x64-windows-static-md\tools\qt5\mkspecs\win32-msvc -ID:\vcpkg\installed\x64-windows-static-md\include\qt5\QtNetwork -ID:\vcpkg\installed\x64-windows-static-md\include\qt5\QtXml -ID:\vcpkg\installed\x64-windows-static-md\include\qt5\QtConcurrent -ID:\vcpkg\installed\x64-windows-static-md\include\qt5\QtSql -ID:\vcpkg\installed\x64-windows-static-md\include\qt5\QtSvg -ID:\vcpkg\installed\x64-windows-static-md\include\qt5\QtWidgets -ID:\vcpkg\installed\x64-windows-static-md\include\qt5\QtGui -ID:\vcpkg\installed\x64-windows-static-md\include\qt5\QtEventDispatcherSupport -ID:\vcpkg\installed\x64-windows-static-md\include\qt5\QtFontDatabaseSupport -ID:\vcpkg\installed\x64-windows-static-md\include\qt5\QtThemeSupport -ID:\vcpkg\installed\x64-windows-static-md\include\qt5\QtAccessibilitySupport -ID:\vcpkg\installed\x64-windows-static-md\include\qt5\QtWindowsUIAutomationSupport /DWIN32 /D_WINDOWS /GR /EHsc /O2 /Ob2 /DNDEBUG -MD /Ox /fp:fast /bigobj /WX -std:c++14 /showIncludes /Fosrc\mumble\CMakeFiles\mumble.dir\PluginManifest.cpp.obj /Fdsrc\mumble\CMakeFiles\mumble.dir\ /FS -c D:\mumble\src\mumble\PluginManifest.cpp
D:\vcpkg\installed\x64-windows-static-md\include\expat_config.h(37): error C2220: the following warning is treated as an error
D:\vcpkg\installed\x64-windows-static-md\include\expat_config.h(37): warning C4005: 'HAVE_STDINT_H': macro redefinition
D:\mumble\src\mumble\PluginManifest.cpp: note: see previous definition of 'HAVE_STDINT_H'
[520/629] Building CXX object src\mumble\CMakeFiles\mumble.dir\PluginManager.cpp.obj
ninja: build stopped: subcommand failed.
 
D:\mumble\build>ni
```


### Screenshots

_No response_

--- a/src/mumble/CMakeLists.txt
+++ b/src/mumble/CMakeLists.txt
@@ -666,20 +666,6 @@ if(bundled-celt)
 	# Disable all warnings that the Celt code may emit
 	disable_warnings_for_all_targets_in("${3RDPARTY_DIR}/celt-0.7.0-build")
 
-	# celt_types.h has some logic of checking whether or not it may include
-	# stdint.h or not. In order to avoid this logic to mess up, we provide
-	# it with the hint that it may indeed include stdint.h and be done with it
-	target_compile_definitions(celt
-		PRIVATE
-			HAVE_STDINT_H
-	)
-	# We also set this flag for Mumble as it includes the problematic
-	# header file indirectly at some points
-	target_compile_definitions(mumble
-		PRIVATE
-			HAVE_STDINT_H
-	)
-
 	add_dependencies(mumble celt)
 
 	target_include_directories(mumble PRIVATE "${3RDPARTY_DIR}/celt-0.7.0-src/libcelt")

